torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -183,6 +187,36 @@ def get_b_layout(self) -> LayoutType:
         return LayoutType.NORMAL
 
 
+class CppMicroGemmF16Abstract(CppMicroGemm):
+    ALLOCATE_BUFFER = r"""
+  int64_t {{buffer_name}}_dtype_itemsize = (
+    std::is_same_v<{{buffer_dtype}}, at::BFloat16>
+    || std::is_same_v<{{buffer_dtype}}, at::Half>
+    ) ? 2 : 4;
+  auto& {{buffer_name}}_allocator = *at::getCPUAllocator();
+  auto {{buffer_name}}_work_data = {{buffer_name}}_allocator.allocate({{buffer_size}}*{{buffer_name}}_dtype_itemsize);
```

Comment:
Heap allocation may be slower than stack allocation (locking overhead due to multi-threading & malloc-internal data structures manipulation. Even tcmalloc isn't completely lock-free) and since such a buffer would be allocated multiple times in multiple threads, can we use a stack allocated buffer, if it results in a non-significant performance difference for large input shapes? Thanks!

The flex attention codegened kernel uses a heap allocator but it's invoked by only one thread, and then individual threads use equal parts of the allocated buffer. Moreover, the buffer it allocates may be large enough not to fit on the default Linux stack size.